### DISCUSSION TITLE
How to get the MAC Address of the client

### DISCUSSION THREAD

**Discussion by mhdcindioglu** [justcoding121/Titanium-Web-Proxy#912] [General]
I need to get the MAC Address of the client.

**Comment by honfika** [comment]
https://stackoverflow.com/questions/1148778/how-do-i-access-arp-protocol-information-through-net

But you can get the MAC address only when the remote machine (client) is in your local network.

**Comment by justcoding121** [comment]
Yep ? what @honfika said. A proxy only sees the TCP connection (IP:port). There's no MAC in HTTP/TLS.

On the same LAN you *might* resolve IP ? MAC via ARP, but that fails across routers/VPN/internet, and isn't something the library exposes.

Use `e.ClientRemoteEndPoint.Address` for the client IP; for MAC you'd need local network tooling outside Titanium.

Closing this one.
